src/DynamoCoreWpf/ViewModels/PackageManager/PublishPackageViewModel.cs

```diff
@@ -931,9 +942,56 @@ private void AddDllFile(string filename)
         /// Delegate used to submit the element</summary>
         private void Submit()
         {
+            var files = BuildPackage();
+            try
+            {
+                // begin submission
+                var pmExtension = dynamoViewModel.Model.GetPackageManagerExtension();
+                var handle = pmExtension.PackageManagerClient.PublishAsync(Package, files, IsNewVersion);
+
+                // start upload
+                Uploading = true;
+                UploadHandle = handle;
+            }
+            catch (Exception e)
+            {
+                ErrorString = e.Message;
+                dynamoViewModel.Model.Logger.Log(e);
+            }
+        }
+
+        /// <summary>
+        /// Delegate used to publish the element to a local folder</summary>
+        private void PublishLocally()
+        {
+            var publishPath = GetPublishFolder();
+            if (string.IsNullOrEmpty(publishPath))
+                return;
+
+            var files = BuildPackage();
+
+            try
+            {
+                UploadState = PackageUploadHandle.State.Copying;
+                Uploading = true;
+                // begin publishing to local directory
+                var remapper = new CustomNodePathRemapper(DynamoViewModel.Model.CustomNodeManager, DynamoModel.IsTestMode);
+                var builder = new PackageDirectoryBuilder(new MutatingFileSystem(), remapper);
+                builder.BuildDirectory(Package, publishPath, files);
+                UploadState = PackageUploadHandle.State.Uploaded;
```

Comment:
From the way it looks I'm not sure if the UI will have a chance to update the status bar...